src/System.Text.Json/tests/NewtonTests/ImmutableCollectionsTests.cs

```diff
@@ -0,0 +1,399 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text.Json.Serialization;
+using Xunit;
+
+namespace System.Text.Json.Tests.Imported
+{
+    public class ImmutableCollectionsTests
+    {
+        private JsonSerializerOptions indentedOption = new JsonSerializerOptions { WriteIndented = true }; 
+
+        #region List
+        [Fact]
+        public void SerializeList()
+        {
+            ImmutableList<string> l = ImmutableList.CreateRange(new List<string>
+            {
+                "One",
+                "II",
+                "3"
+            });
+
+            string json = JsonSerializer.ToString(l);
+            Assert.Equal(@"[""One"",""II"",""3""]", json);
+        }
+
+        [Fact]
+        public void DeserializeList()
+        {
+            string json = @"[
+  ""One"",
+  ""II"",
+  ""3""
+]";
+
+            ImmutableList<string> l = JsonSerializer.Parse<ImmutableList<string>>(json);
+
+            Assert.Equal(3, l.Count);
+            Assert.Equal("One", l[0]);
+            Assert.Equal("II", l[1]);
+            Assert.Equal("3", l[2]);
+        }
+
+        [Fact]
+        public void DeserializeListInterface()
+        {
+            string json = @"[
+        ""Volibear"",
+        ""Teemo"",
+        ""Katarina""
+      ]";
+
+            // what sorcery is this?!
+            IImmutableList<string> champions = JsonSerializer.Parse<IImmutableList<string>>(json);
+
+            Assert.Equal(3, champions.Count);
+            Assert.Equal("Volibear", champions[0]);
+            Assert.Equal("Teemo", champions[1]);
+            Assert.Equal("Katarina", champions[2]);
+        }
+        #endregion
+
+        #region Array
+/*        [Fact]
```

Comment:
Why is this commented out?

Reply:
It is failing with `ImmutableArray` not supported exception. It was my understanding that @layomia is working on this feature and itc going to be supported soon. So wanted to keep the test and uncomment when feature done

Reply:
In this case, I would use `[ActiveIssue()]` attribute on the test with the relevant issue number rather than having commented out code.